### DISCUSSION TITLE
Can Avalonia apps be included in Linux repos?

### DISCUSSION THREAD

**Discussion by mysteryx93** [AvaloniaUI/Avalonia#12023] [General]
Avalonia is great to develop apps for all platforms. For Linux, the most convenient way is to deploy an AppImage which works fine.

However, can Avalonia apps be included in Linux main repositories? The general philosophy is that everything must be open-source and compiled for the specific platform. Avalonia .NET apps, on the other hand, are pre-compiled and compatible for the various platforms.

With that in mind; can those apps be included in the repos or not?

**Comment by kekekeks** [comment]
I'm not quite sure that "every dependency comes from NuGet" model is compatible with "everything should be consuming system-wide installed libraries" model used by most of Linux distros. They also usually require everything to be built from source on their infrastructure, which is troublesome for dependencies like SkiaSharp.

Other languages using their respective dependency packaging systems are suffering from the same problem.

**Reply by mysteryx93** [reply]
and if it's built on native Linux components, then it won't be cross-platform

Although some projects are truly cross-platform. MPV and FFMPEG for example. They're designed on Linux libs, and then the libs also get compiled for Windows? Neither of which have an UI though.

NuGet dependencies however are generally very small; not 1-10MB packages like most Linux packages. Not sure that in itself is an issue.
